File: AlgoExpert/Arrays/01-Two-Number-Sum/Java/Solution.java
import java.util.*;

class Solution {
  public int[] twoNumberSum(int[] array, int targetSum) {
    // Write your code here.
    Set<Integer> numSeen = new HashSet<>();

    for (int number : array) {
      int match = targetSum - number;

      if (numSeen.contains(match)) {
        return new int[]{number, match};
      }
      numSeen.add(number);
    }

    return new int[]{};
  }
}

File: AlgoExpert/Arrays/01-Two-Number-Sum/Golang/solution.go
package main

func TwoNumberSum(array []int, target int) []int {
	// Write your code here.
	numbers_seen := make(map[int]bool) // pseudo set

	for _, number := range array {
		match := target - number

		if _, ok := numbers_seen[match]; ok {
			return []int{match, number}
		}
		numbers_seen[number] = true
	}
	return []int{}
}
